tests/Infrastructure/VersionSyncTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using Xunit;
+
+namespace AzureMcp.Tests.Infrastructure;
+
+public class VersionSyncTests
+{
+    private const string GlobalJsonFileName = "global.json";
+    private const string DockerfileFileName = "Dockerfile";
+    private static readonly string _repoRoot = GetRepoRoot();
+
+    [Fact]
+    public void DotNet_Versions_Should_Be_Synchronized_Between_GlobalJson_And_Dockerfile()
+    {
+        // Arrange
+        var globalJsonPath = Path.Combine(_repoRoot, GlobalJsonFileName);
+        var dockerfilePath = Path.Combine(_repoRoot, DockerfileFileName);
+
+        // Act
+        var globalJsonSdkVersion = GetDotNetSdkVersionFromGlobalJson(globalJsonPath);
+        var dockerfileRuntimeVersion = GetDotNetRuntimeVersionFromDockerfile(dockerfilePath);
+
+        // Assert
+        Assert.True(File.Exists(globalJsonPath), $"{GlobalJsonFileName} not found at {globalJsonPath}");
+        Assert.True(File.Exists(dockerfilePath), $"{DockerfileFileName} not found at {dockerfilePath}");
+        Assert.NotNull(globalJsonSdkVersion);
+        Assert.NotNull(dockerfileRuntimeVersion);
+
+        var sdkVersion = new Version(globalJsonSdkVersion);
+        var runtimeVersion = new Version(dockerfileRuntimeVersion);
+
+        Assert.True(sdkVersion.Major == runtimeVersion.Major && sdkVersion.Minor == runtimeVersion.Minor,
+            $"Major.Minor versions should match between {GlobalJsonFileName} SDK ({sdkVersion}) and {DockerfileFileName} runtime ({runtimeVersion}). " +
+            $"Found SDK: {sdkVersion.Major}.{sdkVersion.Minor}, Runtime: {runtimeVersion.Major}.{runtimeVersion.Minor}");
+    }
+
+    private static string GetDotNetSdkVersionFromGlobalJson(string globalJsonPath)
+    {
+        var jsonContent = File.ReadAllText(globalJsonPath);
+
+        // Remove comments from JSON
```

Comment:
https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsoncommenthandling?view=net-9.0

You can use JsonCommentHandling.Skip instead of this